Documentation/spec/docfx_document_schema.md

```diff
@@ -0,0 +1,197 @@
+>
+> Omitting this keyword has the same behavior as an empty schema.
+
+## 6. Manipulation keywords
+
+### 6.1 behavior
+The value of "behavior" MUST be a string. It defines how DocFX will manipulate the property and affect the build context.
+
+The following values are allowed:
+* `uid`: If the instance is a string, it will be marked as a UID (stands for unique identifier) in the current model, and add it to the document build context.
+* `href`: If the instance is a string and in relative path, it will be converted into path from docset root, and DocFX will update the path afterward.
+* `uidReference`: If the instance is a string, it wlll be taken as a UID, and a depdency on it will be logeged in document build context.
+* `markup`: If the instance is a string, it will be marked up and updated.
+* `include`: if the instance is a string and in relative path, content of the targeted file will replace it in place.
+* `markupInclude`: This is same as `include`, in spite of that the content will be marked up before fill in.
+
+### 6.2 tags
+The value of this keyword MUST be either a string or an array. If it is an array, elements of the array MUST be strings and MUST be unique.
+
+If a processor need to do some operations on a specified property, while the location of the property differs with schemas, it's hard to share the code.
+In this case, a tag can be defined to mark the property, then the processor can fetch the instances with defined tag from schema parser, and read or modify it.
+
+### 6.3 overwrite
+The value of "overwrite" MUST be a string. It specifies how DocFX uses the [overwrite files](..\tutorial\intro_overwrite_files.md) to overwrite the existing values.
+
+The following values are allowed:
+* `replace`: The value will be overwitten by the one specified in overwrite files. This is the default behavior if no overwrite keywords is specified.
+* `key`: If the instance is a string, it will be used as the key of the current object. It can be used to identify the object to overwrite in an array.
+* `merge`: If the instance is an object, it will be merged with the one in overwrite files. If there are duplicated keys, the one in overwrite files will overwrite the existing one.
+* `ignore`: Mark the current instance cannot be overwritten.
+
+> [!Note]
+> After applying the overwrite files, the model should be checked again to ensure that all the validation rules still satifies.
+
+## 7. Undefined keywords
+The schema parser can only recognize the keywords defined above. Undefined keywords is also allowed in the schema. It can provide some description for the schema, or provide more information for other parties to process.
+
+> [!note]
+> As there may be more keywords in JSON schema added to this, it's NOT RECOMMENDED to reuse the keywords already defined in JSON schema here, unless the usage is exactly same.
```

Comment:
change this part to "extension keywords"...

---
In reply to: [122872602](https://github.com/dotnet/docfx/pull/1771#discussion_r122872602) [](ancestors = 122872602)